nerfstudio/exporter/marching_cubes.py

```diff
@@ -0,0 +1,249 @@
+                valid_mask = torch.nn.functional.grid_sample(coarse_mask, pts[None, None, None])[0, 0, 0, 0] > 0
+                if valid_mask.any():
+                    pts_sdf[valid_mask] = evaluate(pts[valid_mask].contiguous())
+            else:
+                pts_sdf = evaluate(pts)
+        else:
+            mask = mask.reshape(-1)
+            pts_to_eval = pts[mask]
+
+            if pts_to_eval.shape[0] > 0:
+                pts_sdf_eval = evaluate(pts_to_eval.contiguous())
+                pts_sdf[mask] = pts_sdf_eval
+
+        if pid < 3:
+            # Update mask
+            mask = torch.abs(pts_sdf) < threshold
+            mask = mask.reshape(coarse_n, coarse_n, coarse_n)[None, None]
+            mask = upsample(mask.float()).bool()
+
+            pts_sdf = pts_sdf.reshape(coarse_n, coarse_n, coarse_n)[None, None]
+            pts_sdf = upsample(pts_sdf)
+            pts_sdf = pts_sdf.reshape(-1)
+
+        threshold /= 2.0
+    return pts_sdf
+
+
+@torch.no_grad()
+def get_surface_sliding(
+    sdf: Callable,
+    output_path: Path,
+    resolution: int = 512,
+    bounding_box_min: Tuple[float, float, float] = (-1.0, -1.0, -1.0),
+    bounding_box_max: Tuple[float, float, float] = (1.0, 1.0, 1.0),
+    level: float = 0,
+    coarse_mask: Union[None, TensorType["height", "width", "depth"]] = None,
+) -> None:
+    """
+    Computes the isosurface of a signed distance function (SDF) defined by the
+    callable `sdf` in a given bounding box with a specified resolution. The SDF
+    is sampled at a set of points within a regular grid, and the marching cubes
+    algorithm is used to generate a mesh that approximates the isosurface at a
+    specified isovalue `level`.
+
+    Args:
+        sdf: A callable function that takes as input a tensor of size
+            (N, 3) containing 3D points, and returns a tensor of size (N,) containing
+            the signed distance function evaluated at those points.
+        output_path: The output directory where the resulting mesh will be saved.
+        resolution: The resolution of the grid used to sample the SDF.
+        bounding_box_min: The minimum coordinates of the bounding box in which the SDF
+            will be evaluated.
+        bounding_box_max: The maximum coordinates of the bounding box in which the SDF
+            will be evaluated.
+        level: The isovalue at which to approximate the isosurface.
+        coarse_mask: A binary mask tensor of size ("height", "width", "depth") that indicates the regions
+            of the bounding box where the SDF is expected to have a zero-crossing. If
+            provided, the algorithm first evaluates the SDF at the coarse voxels where
+            the mask is True, and then refines the evaluation within these voxels using
+            a multi-scale approach. If None, evaluates the SDF at all points in the
+            bounding box.
+    """
+    # Check if resolution is divisible by 512
+    assert resolution % 512 == 0
```

Comment:
Can you throw an error that is descriptive for why "divisible by 512" is important? I see that it's hard-coded below this.

Reply:
So I'm not totally sure why exactly the value 512 was chosen. Its just a fixed value from the original SDFStudio repo for the minimum grid size of (512, 512, 512), I believe it has to do with the way the multi-resolution grid is evaluated so I'll put a comment about that there but I'm not totally sure